### ISSUE TITLE
Imports ignoring some fields during `matched` status

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#7489]
### Summary

TL;DR **When attempting to import some records we'd expect to result in `modified` statuses, instead we're seeing `matched` and new fields are being ignored.**

See also: #7455

When we directly attempt to import a book that already exists (but has new data fields) some imports are "succeeding" but not adding any data.

Minimal repro:

This existing edition has no `authors` or `number_of_pages` https://openlibrary.org/isbn/1577666054.json

We then get the following amazon data:
```
{'title': 'Surface Water-Quality Modeling', 'authors': [{'name': 'Steven C. Chapra'}], 'publish_date': 'Dec 31, 2008', 'source_records': ['amazon:1577666054'], 'number_of_pages': 844, 'publishers': ['Brand: Waveland Pr Inc', 'Waveland Pr Inc'], 'cover': 'https://m.media-amazon.com/images/I/51XNSW275DL._SL500_.jpg', 'isbn_10': ['1577666054'], 'isbn_13': ['9781577666059'], 'physical_format': 'paperback'}
```

As a logged in patron with api privileges (e.g. mekBot) we then run the following javascript to submit this book directly to the importer, which returns success:

```
var bookdata = var bookdata = {'title': 'Surface Water-Quality Modeling', 'authors': [{'name': 'Steven C. Chapra'}], 'publish_date': 'Dec 31, 2008', 'source_records': ['amazon:1577666054'], 'number_of_pages': 844, 'publishers': ['Brand: Waveland Pr Inc', 'Waveland Pr Inc'], 'cover': 'https://m.media-amazon.com/images/I/51XNSW275DL._SL500_.jpg', 'isbn_10': ['1577666054'], 'isbn_13': ['9781577666059'], 'physical_format': 'paperback'}

fetch("https://openlibrary.org/api/import?debug=true", {
    method: 'POST',
    headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
    },
    body: JSON.stringify(bookdata)
}).then(response => response.json()) .then(response => console.log(JSON.stringify(response)))
```

Response:

```
{"success":true,"edition":{"key":"/books/OL26469756M","status":"matched"},"work":{"key":"/works/OL17891805W","status":"matched"}}
```

- **Investigation**

1. upstream defines importapi /api/import that accepts json
2. data is validated before getting to the catalog (using pydantic)
3. eventually catalog.add_book.load() is called
4. load() then calls load_data(rec)

While investigating this code, we noticed that many fields may be disregarded (at least insofar as pertains to a `work`'s state of becoming `matched`)
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/add_book/__init__.py#L642-L656

Here is the code that fetches the existing edition:
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/add_book/__init__.py#L736-L832

EDIT: Actual, I think this is the block we care about (later on)
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/add_book/__init__.py#L797-L822

- **What is wrong?**

We can see two things:
1. that `number_of_pages` is not being checked here.
2. a reminder that `author` is on the work, not the edition (so the pre-check on #7455 may erroneously determine that the edition is missing author, when it does actually exist on the work)

With respect to 1, here we can see only certain fields are being updated and likely need to be extended:
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/add_book/__init__.py#L797-L822

No author has been added to the `work` and the `number_of_pages` did not get added to the `edition`.

- **What caused it?**

Presumably an issue with our catalog import logic for matched editions/works?

Using our new pre-check code for the amz affiliate_server helped surface areas where we may not be importing new available data (resulting in a `matched` response for work and edition, rather than `modified`, e.g. for fields like `number_of_pages`). 

- **What fixed it?**

We identified two further steps:

1. @jimchamp to investigate adding logic to catalog to include fields in our schemata like `number_of_pages`
2. @cclauss to update pre-check to fetch the edition **and** the work (to see if author is needed) 

Step 2 was tested / implemented on a call with @cclauss and patch-deployed for affiliate_server (to decrease NOP imports)

For step 1, we've created a followup P1 issue on our milestone:
https://github.com/internetarchive/openlibrary/issues/7493

- **Followup actions:**

1. Flowchart (a followup of https://github.com/internetarchive/openlibrary/wiki/Developer's-Guide-to-Data-Importing#production-automatic-import-pipeline) which shows how data moves through the catalog / load / load_data.
2. Coming up with a proposed/revised flowchart for what needs to be fixed or re-written. This includes DRYing the pipeline + language for describing how data moves through the system. 

### Steps to close

1. [x] Assignment: Is someone assigned to this issue? (notetaker, responder)
5. [x] Labels: Is there an `Affects:` label applied?
6. [x] Diagnosis: Add a description and scope of the issue
7. [x] Updates: As events unfold, is notable **provenance** documented in issue comments? (i.e. useful debug commands / steps / learnings / reference links)
8. [x] "What caused it?" - please answer in summary
9. [x] "What fixed it?" - please answer in summary
10. [x] "Followup actions:" actions added to summary

**Comment by mekarpeles** [comment]
**Resolution**: A new issue #7493 has been created to address the discovered problem.
